mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp

```diff
@@ -384,15 +404,27 @@ void CHudMapFinishedDialog::SetCurrentPage(int pageNum)
     m_iCurrentPage = pageNum;
 
     wchar_t unicodeTime[BUFSIZETIME];
+    char diffString[BUFSIZETIME];
+    Color diffColor;
+
     //"Time:" shows up when m_iCurrentPage  == 0
     if (m_iCurrentPage < 1)// == 0, but I'm lazy to do an else-if
     {
+        
         m_pCurrentZoneLabel->SetText(m_pwCurrentPageOverall);
 
         MomUtil::FormatTime(m_pRunData ? float(m_pRunData->m_iRunTime) * m_pRunData->m_flTickRate : 0.0f, m_pszEndRunTime);
```

Comment:
I'm noticing a lot of `m_pRunData ? ( . . . ) : 0` stuff here, and that's kinda smelly. Something to investigate is whether `m_pRunData` is actually ever null when going down this codepath, and you can pretty easily check that by doing `Assert( m_pRunData );` at the top of this function. I'm not sure what the use case of calling this function *without* valid run data would be, or if it even happens.

Reply:
I get assertion fails at game launch.
If i remove the check at [line 416](https://github.com/momentum-mod/game/pull/705/files/5056dcf384e2ec9fe378c137a7d39676beabede2#diff-af494e5a9a7c8487930ea44f9e53503dR416) the game crashes.
Removed the other one.

Reply:
That's not good -- this should not be getting updated outside of being in a level. You could use `engine->IsInGame()` to determine that and early return, removing all the `m_pRunData ? (...)  : 0` stuff in this method.

I don't like the null checks everywhere, this panel should not be showing when the run data is null:
https://github.com/momentum-mod/game/blob/13de0cb7711f374791829004311a02806dae1b89/mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp#L186-L189